spec/views_integration/user_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+RSpec.describe User, type: :system do
+  subject { User.new(name: 'Anna', posts_counter: 3, photo: 'https://randomuser.me/api/portraits/women/67.jpg', bio: 'Project manager') }
+  before { subject.save }
+  describe 'index page' do
+    it 'I can see the username of all other users.' do
+      visit root_path(subject)
+      page.has_content?(subject.name)
+    end
+    it 'I can see the profile picture for each user' do
+      visit root_path(subject)
+      page.has_content?(subject.photo)
+    end
+    it 'I can see the number of posts each user has written.' do
+      visit root_path(subject)
+      page.has_content?(subject.posts_counter)
+    end
+    it "When I click on a user, I am redirected to that user's show page." do
+      user2 = User.create(name: 'Lilly', posts_counter: 2, photo: 'https://randomuser.me/api/portraits/women/70.jpg',
+                          bio: 'Teacher from Poland.')
+      visit root_path(user2)
+      click_on 'Lilly'
+      page.has_content?('Lilly')
```

Comment:
- [ ] It would be better to replace `page.has_content?('Lilly')` with `expect(page).to have_current_path("/users/#{@user.id}")` to make the right test case since we have the redirect path

Reply:
Thank you so much for the recommendation, we have made the change at this [commit](https://github.com/Lucash2022/Blog_App/pull/11/commits/409eb10ceecb94744e843eb1d2c7abaae47c0ca4)